x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -244,6 +244,23 @@ public void testUnsupportedAndMultiTypedFields() {
                 + " [ip] in [test1, test2, test3] and [2] other indices, [keyword] in [test6]",
             error("from test* | where multi_typed is not null", analyzer)
         );
+
+        assertEquals(
+            "1:47: Cannot use field [unsupported] with unsupported type [flattened]",
+            error("from test* | eval x = now() + to_timeduration(unsupported)", analyzer)
```

Comment:
nit: the test cases could be parameterized to run for both `to_timeduration` and `to_dateperiod`. Or simply copy-pasted so that both foldables convert functions are tested in each scenario.

Reply:
Done.